examples/unstructured_2d_dgsem/elixir_shallow_water_convergence.jl

```diff
@@ -0,0 +1,69 @@
+
+using Downloads: download
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# semidiscretization of the shallow water equations with a particular
+# bottom topography function
+
+@inline function mms_bottom_topography(x, equations::ShallowWaterEquations2D)
+  # periodic bottom for the periodic with a twist mesh domain [0, sqrt(2)]^2
+  return 2.0 + 0.5 * sin(sqrt(2.0)*pi*x[1]) + 0.5 * sin(sqrt(2.0)*pi*x[2])
+end
+
+equations = ShallowWaterEquations2D(9.81, bottom_topography=mms_bottom_topography)
+
+initial_condition = initial_condition_convergence_test
+
+###############################################################################
+# Get the DG approximation space
+
+volume_flux = (flux_wintermeyer_etal, flux_nonconservative_shallow_water_volume)
+solver = DGSEM(polydeg=6, surface_flux=(flux_fjordholm_etal, flux_nonconservative_wintermeyer),
```

Comment:
Why did you choose these fluxes? Well, I can of course guess why, but since we're now several years past your paper, we can reconsider these choices. Why shouldn't we use the same flux in the volume and at the interfaces? Why do we use these two specific fluxes instead of any other flux of the (at least) two-parameter family of EC fluxes?

Reply:
I chose the volume flux from the paper with Niklas because I know that it is high-order, ec, and well-balanced. The surface nonconservative flux was selected because I know it works on curved meshes even with discontinuous bottom topographies. Also, it helps in debugging because my Fortran code uses these.

Something in the future will be to put in other EC/ES fluxes. This is especially true for the nonconservative coupling that is also well-balanced. There are many more recent schemes that improve upon the original hydrostatic reconstruction idea from Audusse et al. In particular, I am talking about [Chen & Noelle](https://epubs.siam.org/doi/abs/10.1137/15M1053074) or [Buttinger-Kreuzhuber et al.](https://www.sciencedirect.com/science/article/pii/S0309170818305335).